druid/src/widget/flex.rs

```diff
@@ -37,6 +38,26 @@ pub(crate) enum Axis {
     Vertical,
 }
 
+/// The alignment of the widgets inside of the container.
+///
+/// If a widget is smaller than the container on the minor axis, this determines
+/// where it is positioned.
+#[derive(Debug, Clone, Copy)]
+pub enum Alignment {
+    /// Top or leading.
+    ///
+    /// In a vertical container, widgets are top aligned. In a horiziontal
+    /// container, their leading edges are aligned.
+    First,
```

Comment:
This feels weird. Would something like this be possible or even desirable?
```rust
pub enum VerticalAlignment {
    Top,
    Center,
    End,
}

pub enum HorizontalAlignment {
    Leading,
    Center,
    Trailing,
}
```
Then modify `Axis`
```rust
pub(crate) enum Axis {
    Horizontal(HorizontalAlignment),
    Vertical(VerticalAlignment),
}
```
This could be even worse UX wise, just tossing around an idea :man_shrugging:

Reply:
heh I actually had almost literally that code, except with still having my `Alignment` type internally, and using `From<VerticalAlignment>` and `From<HorizontalAlignment>` for it. The problem I ran into is that the vertical/horizontal axis is not really enforced in the type system; that is to say, I can't at compile time ensure that a vertical `Flex` can accept a `HorizontalAlignment` but not a `VerticalAlignment`.

The one way I could think of would be to *require* the `row()` or `column()` constructor to take an explicit vertical or horizontal alignment argument, but this breaks a very common api in a situation where there is probably a reasonable default.

So I ended up with this.

In a funny way, making this change does start to argue for having separate `Row`/`Column` or `HStack`/`VStack` types, because then we could tie the allowable alignment to the type system. As is, though, I really couldn't find a satisfying solution.

Reply:
Yeah I was trying to avoid suggesting separate vertical and horizontal versions of the struct :thinking: Thankfully `Flex` doesn't seem to have more than ~120 LOC will eventually lessen the pain of duplicating most of it....

Reply:
Oh, we wouldn't need to duplicate it I don't think, we could just use it as private API, with types like:


```rust
pub struct Row<T> {
    inner: Flex<T>,
    alignment: HorizontalAlignment,
}

pub struct Column<T> {
    inner: Flex<T>,
    alignment: VerticalAlignment,
}
```

Reply:
but again I don't love this. Another option would be to have `Leading`, `Top`, `Trailing`, `Bottom`, be essentially aliases for each other. 🤷‍♂

Reply:
Oh nice! That makes me feel better about that lol

Reply:
In any case my inclination is to leave this as-is until at least we implement baseline alignment, and then we can consider switching the API again?